services/shhext/api.go

```diff
@@ -186,7 +188,44 @@ func (api *PublicAPI) GetNewFilterMessages(filterID string) ([]*whisper.Message,
 	if err != nil {
 		return nil, err
 	}
-	return api.service.deduplicator.Deduplicate(msgs), err
+
+	dedupMessages := api.service.deduplicator.Deduplicate(msgs)
+
+	// Attempt to decrypt message, otherwise leave unchanged
+	for _, msg := range dedupMessages {
+		var privateKey *ecdsa.PrivateKey
+		var publicKey *ecdsa.PublicKey
+
+		// Msg.Dst is empty is a public message, nothing to do
+		if msg.Dst != nil {
+			// There's probably a better way to do this
+			keyBytes, err := hexutil.Bytes(msg.Dst).MarshalText()
+			if err != nil {
+				return nil, err
+			}
+
+			privateKey, err = api.service.w.GetPrivateKey(string(keyBytes))
```

Comment:
@cammellos  as I understand `keyBytes` is the public key, can you call `GetPrivateKey` with that? I thought you had to pass the id but maybe I'm missing something

Reply:
ah I found this in the docs:

```
// Originally, public keys
// were used as keys, now random keys are being used. And in order to
// make it easier to consume, we now allow both random IDs and public
// keys to be passed.
func toDeterministicID(id string, expectedLen int) (string, error) {
```

but does it work with public key for a key created previously?

Reply:
It seems to be working for older clients ( I have them successfully communicate over public and 1-to-1 chats), but what do you mean exactly created previously? How long ago? (Status-react only stores the key as pk, not the ID)